sqlmesh/core/model/definition.py

```diff
@@ -615,8 +615,9 @@ def is_seed(self) -> bool:
 
     @cached_property
     def depends_on_past(self) -> bool:
-        if self.kind.is_incremental_by_unique_key:
-            return True
+        explicit_depends_on_past = getattr(self.kind, "depends_on_past", None)
```

Comment:
note that my original implementation did something like:

```
if self.kind.is_incremental and self.kind.depends_on_past is not None:
  return self.kind.depends_on_past
```

but this caused all the `ScdType2Kind` tests to fail because those `Kind`'s still return `True` from the `is_incremental` check but they dont inherit from `_Incremental`